src/Application/Actions/Donations/Update.php

```diff
@@ -258,10 +272,35 @@ private function addData(Donation $donation, HttpModels\Donation $donationData,
             }
 
             if ($donationData->autoConfirmFromCashBalance) {
-                $this->stripeClient->paymentIntents->confirm($donation->getTransactionId());
+                try {
+                    $this->stripeClient->paymentIntents->confirm($donation->getTransactionId());
+                } catch (InvalidRequestException $exception) {
+                    // Currently a typical Update call which auto-confirms is being made for just
+                    // that purpose. So our safest options are to return a 500 and roll back any
+                    // database changes.
+                    $this->entityManager->rollback();
+
+                    // To help analyse it quicker we handle the specific auto-confirm API failure we've
+                    // seen before with a distinct message, but both options give the client an HTTP 500,
+                    // as we expect neither with our updated guard conditions.
+                    if (str_contains($exception->getMessage(), 'missing a payment method')) {
```

Comment:
Maybe worth making this a bit more specific, to make sure we don't catch any similar but different errors here?

```suggestion
                    if (str_starts_with($exception->getMessage(), "You cannot confirm this PaymentIntent because it's missing a payment method")) {

```